### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by wprl** [wprl/seneca-cluster-test#1]
@vertiman let's move discussion here to keep everything in one place if that is OK with you. 

Since this example seems to be working, I think there are a few possibilities:

1. This example code is too simplistic so it fails to reproduce the timeouts somehow.
2. There is a difference between my development environment and yours, in which case you would see timeouts running this code.
3. The invalid_origin warnings are a red herring and the timeouts are being caused by other code.

Would it be possible to see your code or a simplified version of it that reproduces the timeouts?  If you wouldn't mind checking if you get timeouts when you run this code, that would also help me diagnose this.

cc: @rjrodger

**Comment by vertiman** [comment]
happy to put together some code.  unfortunately i won't have time until
tomorrow.  I'll throw some details your way in a bit.

**Comment by vertiman** [comment]
I figured out what's going on - it's in the nature of the seneca beanstalk transport implementation that it works.  In the unpatched RabbitMQ transport it fails.  

Let's say we have Client A, Client B and Server A.

Client A puts message into request queue.  Server A picks up message, processes, responds.  At this point whether Client A or Client B pick up the response is dependent on whoever gets to the queue first.  If Client B picks up the message, and INVALID_ORIGIN message is popped up and Seneca does not process.

That is the point where Rabbit and Beanstalk differ.  With beanstalk, the transport calls release instead of destroy and that causes the message to be available in the queue again.  Eventually, each message will be picked up by the right client but it might be many tries until it does.

With Rabbit, it appears that messages are ack'd immediately upon receipt and without checking for error state, so the messages just get flagged as complete.  So I'd say this is a bug in the Rabbit transport, and not Seneca itself. 

tl;dr - change to rabbitmq-transport and you'll see the issue in your example!  :)

**Comment by vertiman** [comment]
PS - it's' also way easier to see the issue with a tool like siege.

siege -c 100 -q http://localhost:3000/ping

**Comment by wprl** [comment]
Yes, you're exactly right.  I've updated this repo with working examples for both transports and created a PR for seneca-beanstalk-transport using the basic idea from the other PR.  Nice tip about siege :)
